diff --git a/setup.py b/setup.py
index 1842198..9fe272a 100644
--- a/setup.py
+++ b/setup.py
@@ -11,7 +11,8 @@ from setuptools import setup
 
 def get_extensions():
     import numpy as np
-    from extension_helpers import add_openmp_flags_if_available, pkg_config
+    from extension_helpers import pkg_config
+    from extension_helpers._openmp_helpers import get_openmp_flags
     from setuptools import Extension
 
     pkg_config_packages = ["gsl"]
@@ -56,7 +57,9 @@ def get_extensions():
     )
 
     if not os.environ.get("LIGO_SKYMAP_DISABLE_OPENMP"):
-        add_openmp_flags_if_available(extension)
+        openmp_flags = get_openmp_flags()
+        extension.extra_compile_args.extend(openmp_flags.get('compiler_flags'))
+        extension.extra_link_args.extend(openmp_flags.get('linker_flags'))
 
     return [extension]
 
