diff --git a/pyproject.toml b/pyproject.toml
index fef2b0d..e6e008a 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -38,7 +38,6 @@ dependencies = [
     "h5py",
     "igwn-ligolw",
     "igwn-segments",
-    "lalsuite>=7.26",  # https://git.ligo.org/lscsoft/lalsuite/-/merge_requests/2397
     "ligo-gracedb>=2.0.1",
     "matplotlib>=3.9.1",  # https://github.com/matplotlib/matplotlib/pull/28289
     "networkx",
