diff --git a/src/bayestar_moc.c b/src/bayestar_moc.c
index 6042633..e3ef87e 100644
--- a/src/bayestar_moc.c
+++ b/src/bayestar_moc.c
@@ -46,6 +46,18 @@ int8_t uniq2order64(int64_t uniq)
         : "=r" (o)
         : "rm" (uniq));
     order = o;
+#elif defined(__aarch64__)
+    int64_t o;
+    asm("clz %0, %1\n\t"
+        : "=r" (o)
+        : "r" (uniq));
+    order = 63 - o;
+#elif defined(__powerpc64__)
+    int64_t o;
+    asm("cntlzd %0, %1\n\t"
+        : "=r" (o)
+        : "r" (uniq));
+    order = 63 - o;
 #else
     for (order = -1; uniq; uniq >>= 1, order ++)
         /* noop */;
